test/spec/Feature/Auth/AudienceJwtSecretSpec.hs

```diff
@@ -3,43 +3,242 @@ module Feature.Auth.AudienceJwtSecretSpec where
+              "aud": ""
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
+
+  context "when the audience claim is an array of strings" $ do
+    it "succeeds when the audience claim has 1 element and it matches" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["youraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "succeeds when the audience claim has more than 1 element and one matches" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience", "youraudience", "anotheraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith` 200
+
+    it "fails when the audience claim has 1 element and it doesn't match" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
+
+
+    it "fails when the audience claim has more than 1 element and none matches" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": ["notyouraudience", "stillnotyouraudience", "anotheraudience"]
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
+
+    it "fails when the audience claim is empty" $ do
+      let jwtPayload = [json|
+            {
+              "exp": 9999999999,
+              "role": "postgrest_test_author",
+              "id": "jdoe",
+              "aud": []
+            }|]
+          auth = authHeaderJWT $ generateJWT jwtPayload
+      request methodGet "/authors_only" [auth] ""
+        `shouldRespondWith`
+          [json|{"code":"PGRST303","details":null,"hint":null,"message":"JWT not in audience"}|]
+          { matchStatus = 401 }
```

Comment:
This also fails in `v12`, but should we keep failing here? In this case `jwt-aud = youraudience` and the `aud` claim has an empty array `[]`. The logic is that the `aud` is "present" and our audience does not match, so we reject it. Or could we interpret it as "there's no audience" and allow the JWT? The RFC seems to be flexible here, I think:

> The interpretation of audience values is generally application specific.

Unless this means that **once** the `aud` claim is validated we can chose what to do with it?

Reply:
@laurenceisla Since we have:

> The interpretation of audience values is generally application specific.

How about documenting our interpretation? I think more docs about `aud` would fit in https://docs.postgrest.org/en/v12/references/auth.html#jwt-claims-validation

Reply:
It looks like most of the info for `aud` is [in the configuration page](https://postgrest--4043.org.readthedocs.build/en/4043/references/configuration.html#jwt-aud). I added the interpretation there.